quadtree/src/lib.rs

```diff
@@ -55,11 +55,19 @@ pub struct Node {
 }
 
 impl Node {
-    pub fn root_with_bounding_rect(rect: Rect) -> Self {
-        Node {
-            id: NodeId::root(),
+    pub fn from_node_id_and_root_bounding_rect(id: NodeId, rect: Rect) -> Self {
+        let mut ids = vec![id];
+        while let Some(parent_id) = ids.last().and_then(|last| last.parent_id()) {
+            ids.push(parent_id);
+        }
+        let mut node = Self {
+            id: ids.pop().unwrap(),
             bounding_rect: rect,
+        };
+        while let Some(child_id) = ids.pop().and_then(|id| id.child_index()) {
```

Comment:
(assuming that root is not in `ids` so `child_id.child_index().unwrap()` should be safe) We could use iterators here:
`ids.iter().rev().fold(node, |node, child_id| node.get_child(&child_id.child_index().unwrap()))`
not sure if it makes it cleaner though.